cluster_management/src/main/java/com/pinterest/rocksplicator/shardmap/Instance.java

```diff
@@ -0,0 +1,64 @@
+package com.pinterest.rocksplicator.shardmap;
+
+import java.util.Objects;
+
+public class Instance {
+  private final String host;
+  private final int port;
+  private final String domain;
+  private final String instanceId;
+
+  public Instance(String hostWithDomainName) {
+    String[] parts = hostWithDomainName.split(":");
+    this.host = parts[0];
+    this.port = Integer.parseInt(parts[1]);
+    this.domain = parts[2];
+    this.instanceId = String.format("%s_%d", this.host, this.port);
+  }
+
+  public String getInstanceId() {
+    return instanceId;
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  public int getPort() {
+    return port;
+  }
+
+  public String getDomain() {
+    return domain;
+  }
+
+  @Override
```

Comment:
why we have Override here? Instance seem not inherit from other class.

Reply:
All Java classes implicitly extends Object class which is root of all java classes. This method is being overridden so that this works with containers like Map/Set.